### PR TITLE
[WIP] Use config file to bootstrap a game

### PR DESCRIPTION
**Pull request by xtuc**
Launch arena with the following command:

```sh
go build && ./arena config.json
```

### CODE REVIEW

File: cmd/arena/config.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "agents": [
+    {
+      "cmd": "node --harmony /scripts/client.js"
+    },
+    {
+      "cmd": "node --harmony /scripts/client.js"
+    }
+  ]
+}
```
Comment (xtuc) [inline cmd/arena/config.json]: I would prefer using YAML with support for inheritance.

Configuration could look like that:
```yaml
agent: &default
  cmd: "node --harmony /scripts/client.js"

agents:
  foo:
    <<: *default
  bar:
    <<: *default
```

File: server/protocol/config.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package protocol
+
+type Config struct {
+	Cmd string
+}
+
+type FileConfigWrapper struct {
+	Agents []Config
+}
```
Comment (xtuc) [inline server/protocol/config.go]: For now the configuration is very simple. We can use it to bootstrap game mods.

Agent configuration must be resolved separately of server's configuration.

Config type shouldn't be part of the protocol, this is temporary.

File: cmd/arena/main.go [hunk 3]
```diff
@@ -117,8 +120,21 @@ func main() {
 	agentstate.Radius = 16
 	srv.RegisterAgent(attractoragent.MakeAttractorAgent(), agentstate)
 
-	for i := 0; i < cmdenv.agents; i++ {
-		go srv.Spawnagent()
+	filename := os.Args[1]
```
Comment (xtuc) [inline cmd/arena/main.go]: The file location is the first argument

File: cmd/arena/config.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "server": {
+    "port": 8080,
+    "agentdir": "../../agents/",
+    "host": "192.168.1.19",
+    "tps": 20
+  },
+  "agents": [{
+    "scale": 10,
```
Comment (netgusto) [inline cmd/arena/config.json:9]: Scale ? C'est pas plutôt "number" ? ou "instances" ?

Comment (xtuc) [reply cmd/arena/config.json:9]: docker-compose termonology, like scale this container x times

File: server/config/config.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+type fileServerConfig struct {
+	Server struct {
+		Port     int
+		Tps      int
+		Host     string
+		Agentdir string
+	}
+	Agents []struct {
+		Scale int
```
Comment (netgusto) [inline server/config/config.go:34]: Idem. Scale ?

File: cmd/arena/config.json (hunk 4 above)
Comment (netgusto) [reply cmd/arena/config.json:9]: ok, didn't knew that.

**Comment by netgusto** [comment]
LGTM.

File: server/container/orchestrator.go [hunk 6]
```diff
@@ -88,9 +88,9 @@ func (orch *ContainerOrchestrator) TearDownAll() {
 func (orch *ContainerOrchestrator) CreateAgentContainer(agentid uuid.UUID, host string, port int, agentdir string, config config.AgentGameConfig) (AgentContainer, error) {
 
 	containerconfig := container.Config{
-		Image: "node",
+		Image: config.Image,
 		Cmd:   []string{"/bin/bash", "-c", config.Cmd},
-		User:  "node",
+		User:  "root",
```
Comment (netgusto) [inline server/container/orchestrator.go]: TODO: harden container (see issue #3) and avoid using `root`.

**Pull request merged**